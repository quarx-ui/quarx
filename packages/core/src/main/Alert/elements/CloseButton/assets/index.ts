import { ReactComponent as CrossIconSmall } from './cross-small.svg';
import { ReactComponent as CrossIcon } from './cross.svg';

export {
    CrossIconSmall,
    CrossIcon,
};
