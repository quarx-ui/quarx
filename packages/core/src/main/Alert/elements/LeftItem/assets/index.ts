import { ReactComponent as LightningIcon } from './lightning.svg';
import { ReactComponent as EnvelopeIconSmall } from './envelope-small.svg';
import { ReactComponent as InfoIcon } from './info.svg';
import { ReactComponent as CheckmarkCircleIconSmall } from './checkmark-circle-small.svg';
import { ReactComponent as AttentionTriangleIconSmall } from './attention-triangle-small.svg';
import { ReactComponent as AttentionRhombusIconSmall } from './attention-rhombus-small.svg';
import { ReactComponent as DocumentTextIcon } from './document-text.svg';
import { ReactComponent as EnvelopeIconLarge } from './envelope-large.svg';
import { ReactComponent as AttentionCircleIcon } from './attention-circle.svg';
import { ReactComponent as CheckmarkCircleIconLarge } from './checkmark-circle-large.svg';
import { ReactComponent as AttentionTriangleIconLarge } from './attention-triangle-large.svg';
import { ReactComponent as AttentionRhombusIconLarge } from './attention-rhombus-large.svg';

export {
    LightningIcon,
    EnvelopeIconSmall,
    InfoIcon,
    CheckmarkCircleIconSmall,
    AttentionTriangleIconSmall,
    AttentionRhombusIconSmall,
    DocumentTextIcon,
    EnvelopeIconLarge,
    AttentionCircleIcon,
    CheckmarkCircleIconLarge,
    AttentionTriangleIconLarge,
    AttentionRhombusIconLarge,
};
