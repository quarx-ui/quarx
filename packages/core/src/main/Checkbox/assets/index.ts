import { ReactComponent as CheckIconSmall } from './checkboxIconSmall.svg';
import { ReactComponent as CheckIconMedium } from './checkboxIconMedium.svg';
import { ReactComponent as CheckIconLarge } from './checkboxIconLarge.svg';

export { CheckIconLarge, CheckIconSmall, CheckIconMedium };
