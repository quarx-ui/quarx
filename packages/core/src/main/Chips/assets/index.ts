import { FC, SVGProps } from 'react';
import { ChipsSize } from '@core';
import { ReactComponent as CheckMarkIconSmallSize } from './small/checkmarkIcon.svg';
import { ReactComponent as CheckMarkIconMediumSize } from './medium/checkmarkIcon.svg';
import { ReactComponent as ClosingIconSmallSize } from './small/closeIcon.svg';
import { ReactComponent as ClosingIconMediumSize } from './medium/closeIcon.svg';
import { ReactComponent as TriangleIconSmallSize } from './small/triangleIcon.svg';
import { ReactComponent as TriangleIconMediumSize } from './medium/triangleIcon.svg';

export const CheckMarkIcon: Record<ChipsSize, FC<SVGProps<SVGElement>>> = {
    small: CheckMarkIconSmallSize,
    medium: CheckMarkIconMediumSize,
};

export const ClosingIcon: Record<ChipsSize, FC<SVGProps<SVGElement>>> = {
    small: ClosingIconSmallSize,
    medium: ClosingIconMediumSize,
};

export const TriangleIcon: Record<ChipsSize, FC<SVGProps<SVGElement>>> = {
    small: TriangleIconSmallSize,
    medium: TriangleIconMediumSize,
};
