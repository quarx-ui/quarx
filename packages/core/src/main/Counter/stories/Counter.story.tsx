import React from 'react';
import { BASE_ARG_TYPES } from '@core/storybook/BASE_ARG_TYPES';
import { Story } from '@storybook/react/types-6-0';
import { excludeProp } from '@core/storybook/templateParams';
import { DisplayVariants } from '@core/storybook/DisplayVariants';
import { STORY_PATHS } from '@quarx-ui/storybook/utils';
import { Counter, CounterColor, CounterProps, CounterSize, CounterType } from '..';

const defaultArgs: CounterProps = {
    children: 9,
    size: 'large',
    type: 'filled',
    color: 'brand',
    maxDigits: 2,
};

export default {
    title: STORY_PATHS.core.components.main('Counter'),
    component: Counter,
    argTypes: {
        children: {},
        size: {},
        type: {},
        color: {},
        maxDigits: {},
        ...excludeProp(['permissions'], BASE_ARG_TYPES),
    },
    parameters: {
        actions: { disable: true },
    },
    args: defaultArgs,
};

const Template: Story<CounterProps> = (props) => <Counter {...props} />;

export const Sandbox = Template.bind({});
Sandbox.storyName = 'Компонент';

const TYPES: CounterType[] = ['filled', 'ghosted', 'white'];
const COLORS: CounterColor[] = ['brand', 'secondary', 'warning', 'danger', 'info', 'text'];
const SIZES: CounterSize[] = ['small', 'medium', 'large'];

export const Types: Story<CounterProps> = (props) => DisplayVariants({
    property: 'type',
    values: TYPES,
    component: Counter,
    componentProps: props,
});

Types.storyName = 'Типы';
Types.argTypes = excludeProp(['type']);

export const Colors: Story<CounterProps> = (props) => DisplayVariants({
    property: 'color',
    values: COLORS,
    component: Counter,
    componentProps: props,
});

Colors.storyName = 'Цвета';
Colors.argTypes = excludeProp(['color']);

export const Sizes: Story<CounterProps> = (props) => DisplayVariants({
    property: 'size',
    values: SIZES,
    component: Counter,
    componentProps: props,
});

Sizes.storyName = 'Размеры';
Sizes.argTypes = excludeProp(['size']);

const variantsArray = [
    Types,
    Colors,
    Sizes,
];

variantsArray.forEach((variant, index) => {
    variantsArray[index].args = defaultArgs;
});
