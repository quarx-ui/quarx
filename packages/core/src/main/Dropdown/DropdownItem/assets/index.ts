import { ReactComponent as ActiveIcon } from './ActiveIcon.svg';

export { ActiveIcon };
