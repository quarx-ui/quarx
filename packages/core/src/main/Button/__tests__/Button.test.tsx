import { FC, ReactNode } from 'react';
import { cleanup, render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { testStyleParams } from '@core/test-utils';
import { Button, IconButton, ButtonStyleParams, ButtonProps, IconButtonProps } from '@core';
import { PaperClipIcon } from '../stories/assets';

const checkComponent = (Component: FC<ButtonProps & IconButtonProps>, children: ReactNode) => {
    testStyleParams<Partial<ButtonStyleParams>, ButtonProps & IconButtonProps>(
        Component,
        {
            size: 'medium',
            borderRadius: 'medium',
            color: 'brand',
            type: 'contained',
        },
        { children: 'Button' },
    )({
        color: ['brand', 'secondary', 'info', 'success', 'warning', 'danger'],
        size: ['xSmall', 'small', 'medium', 'large'],
        borderRadius: ['xSmall', 'small', 'medium', 'large', 'xLarge', 'max'],
        type: ['contained', 'outlined', 'text'],
        loading: [true, false],
    });

    it('loading should be in the document', () => {
        render(<Component loading>{children}</Component>);
        const button = screen.getByRole('button');
        const loader = button.children[1];

        expect(loader).toBeInTheDocument();
        expect(loader).toHaveClass('QxBaseButton-loader');
        expect(button).toHaveAttribute('disabled');
    });

    it('button should have attribute disabled', () => {
        const mockedOnClick = jest.fn();
        render(<Component disabled onClick={mockedOnClick}>{children}</Component>);
        const button = screen.getByRole('button');

        expect(button).toHaveAttribute('disabled');
        userEvent.click(button);
        expect(mockedOnClick).not.toBeCalled();
    });

    it('button should be hidden', () => {
        render(<Component hidden>{children}</Component>);
        expect(screen.queryByRole('button')).not.toBeInTheDocument();

        cleanup();

        render(<Component permissions={{ hidden: true }}>{children}</Component>);
        expect(screen.queryByRole('button')).not.toBeInTheDocument();
    });

    it('onClick should be called', () => {
        const mockedOnClick = jest.fn();
        render(<Component onClick={mockedOnClick}>{children}</Component>);
        const button = screen.getByRole('button');

        userEvent.click(button);
        expect(mockedOnClick).toBeCalledTimes(1);
    });
};

describe('Button', () => {
    it('text should be in the document', () => {
        render(<Button>Button</Button>);

        expect(screen.queryByText('Button')).toBeInTheDocument();
        expect(screen.getByRole('button')).not.toHaveAttribute('disabled');
    });
    Button.displayName = 'Button';
    checkComponent(Button, 'Button');
});

describe('IconButton', () => {
    it('svg should be in the document', () => {
        render(<IconButton><PaperClipIcon /></IconButton>);
        const svgElement = document.querySelector('svg');
        expect(svgElement).toBeInTheDocument();
    });
    IconButton.displayName = 'IconButton';
    checkComponent(IconButton, <PaperClipIcon />);
});
