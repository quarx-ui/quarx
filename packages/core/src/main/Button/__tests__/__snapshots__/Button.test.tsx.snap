// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button style params borderRadius large 1`] = `
<DocumentFragment>
  <button
    class="QxBaseButton QxBaseButton_size_medium QxBaseButton_color_brand QxBaseButton_type_contained QxBaseButton_borderRadius_large QxButton QxButton_size_medium css-v0k52e-QxBaseButton-root-QxButton-root"
    data-testid="testStyleParams-borderRadius-large"
    type="button"
  >
    <div
      class="QxButton-content css-m3eosc-QxButton-content"
    >
      Button
    </div>
  </button>
</DocumentFragment>
`;

exports[`Button style params borderRadius max 1`] = `
<DocumentFragment>
  <button
    class="QxBaseButton QxBaseButton_size_medium QxBaseButton_color_brand QxBaseButton_type_contained QxBaseButton_borderRadius_max QxButton QxButton_size_medium css-1uhzml3-QxBaseButton-root-QxButton-root"
    data-testid="testStyleParams-borderRadius-max"
    type="button"
  >
    <div
      class="QxButton-content css-m3eosc-QxButton-content"
    >
      Button
    </div>
  </button>
</DocumentFragment>
`;

exports[`Button style params borderRadius medium 1`] = `
<DocumentFragment>
  <button
    class="QxBaseButton QxBaseButton_size_medium QxBaseButton_color_brand QxBaseButton_type_contained QxBaseButton_borderRadius_medium QxButton QxButton_size_medium css-xf29is-QxBaseButton-root-QxButton-root"
    data-testid="testStyleParams-borderRadius-medium"
    type="button"
  >
    <div
      class="QxButton-content css-m3eosc-QxButton-content"
    >
      Button
    </div>
  </button>
</DocumentFragment>
`;

exports[`Button style params borderRadius small 1`] = `
<DocumentFragment>
  <button
    class="QxBaseButton QxBaseButton_size_medium QxBaseButton_color_brand QxBaseButton_type_contained QxBaseButton_borderRadius_small QxButton QxButton_size_medium css-irifgu-QxBaseButton-root-QxButton-root"
    data-testid="testStyleParams-borderRadius-small"
    type="button"
  >
    <div
      class="QxButton-content css-m3eosc-QxButton-content"
    >
      Button
    </div>
  </button>
</DocumentFragment>
`;

exports[`Button style params borderRadius xLarge 1`] = `
<DocumentFragment>
  <button
    class="QxBaseButton QxBaseButton_size_medium QxBaseButton_color_brand QxBaseButton_type_contained QxBaseButton_borderRadius_xLarge QxButton QxButton_size_medium css-7af60g-QxBaseButton-root-QxButton-root"
    data-testid="testStyleParams-borderRadius-xLarge"
    type="button"
  >
    <div
      class="QxButton-content css-m3eosc-QxButton-content"
    >
      Button
    </div>
  </button>
</DocumentFragment>
`;

exports[`Button style params borderRadius xSmall 1`] = `
<DocumentFragment>
  <button
    class="QxBaseButton QxBaseButton_size_medium QxBaseButton_color_brand QxBaseButton_type_contained QxBaseButton_borderRadius_xSmall QxButton QxButton_size_medium css-1q19eh3-QxBaseButton-root-QxButton-root"
    data-testid="testStyleParams-borderRadius-xSmall"
    type="button"
  >
    <div
      class="QxButton-content css-m3eosc-QxButton-content"
    >
      Button
    </div>
  </button>
</DocumentFragment>
`;

exports[`Button style params color brand 1`] = `
<DocumentFragment>
  <button
    class="QxBaseButton QxBaseButton_size_medium QxBaseButton_color_brand QxBaseButton_type_contained QxBaseButton_borderRadius_medium QxButton QxButton_size_medium css-xf29is-QxBaseButton-root-QxButton-root"
    data-testid="testStyleParams-color-brand"
    type="button"
  >
    <div
      class="QxButton-content css-m3eosc-QxButton-content"
    >
      Button
    </div>
  </button>
</DocumentFragment>
`;

exports[`Button style params color danger 1`] = `
<DocumentFragment>
  <button
    class="QxBaseButton QxBaseButton_size_medium QxBaseButton_color_danger QxBaseButton_type_contained QxBaseButton_borderRadius_medium QxButton QxButton_size_medium css-1s8avop-QxBaseButton-root-QxButton-root"
    data-testid="testStyleParams-color-danger"
    type="button"
  >
    <div
      class="QxButton-content css-m3eosc-QxButton-content"
    >
      Button
    </div>
  </button>
</DocumentFragment>
`;

exports[`Button style params color info 1`] = `
<DocumentFragment>
  <button
    class="QxBaseButton QxBaseButton_size_medium QxBaseButton_color_info QxBaseButton_type_contained QxBaseButton_borderRadius_medium QxButton QxButton_size_medium css-c11cza-QxBaseButton-root-QxButton-root"
    data-testid="testStyleParams-color-info"
    type="button"
  >
    <div
      class="QxButton-content css-m3eosc-QxButton-content"
    >
      Button
    </div>
  </button>
</DocumentFragment>
`;

exports[`Button style params color secondary 1`] = `
<DocumentFragment>
  <button
    class="QxBaseButton QxBaseButton_size_medium QxBaseButton_color_secondary QxBaseButton_type_contained QxBaseButton_borderRadius_medium QxButton QxButton_size_medium css-2ejju1-QxBaseButton-root-QxButton-root"
    data-testid="testStyleParams-color-secondary"
    type="button"
  >
    <div
      class="QxButton-content css-m3eosc-QxButton-content"
    >
      Button
    </div>
  </button>
</DocumentFragment>
`;

exports[`Button style params color success 1`] = `
<DocumentFragment>
  <button
    class="QxBaseButton QxBaseButton_size_medium QxBaseButton_color_success QxBaseButton_type_contained QxBaseButton_borderRadius_medium QxButton QxButton_size_medium css-zvyu5w-QxBaseButton-root-QxButton-root"
    data-testid="testStyleParams-color-success"
    type="button"
  >
    <div
      class="QxButton-content css-m3eosc-QxButton-content"
    >
      Button
    </div>
  </button>
</DocumentFragment>
`;

exports[`Button style params color warning 1`] = `
<DocumentFragment>
  <button
    class="QxBaseButton QxBaseButton_size_medium QxBaseButton_color_warning QxBaseButton_type_contained QxBaseButton_borderRadius_medium QxButton QxButton_size_medium css-1ts9iyx-QxBaseButton-root-QxButton-root"
    data-testid="testStyleParams-color-warning"
    type="button"
  >
    <div
      class="QxButton-content css-m3eosc-QxButton-content"
    >
      Button
    </div>
  </button>
</DocumentFragment>
`;

exports[`Button style params loading false 1`] = `
<DocumentFragment>
  <button
    class="QxBaseButton QxBaseButton_size_medium QxBaseButton_color_brand QxBaseButton_type_contained QxBaseButton_borderRadius_medium QxButton QxButton_size_medium css-xf29is-QxBaseButton-root-QxButton-root"
    data-testid="testStyleParams-loading-false"
    type="button"
  >
    <div
      class="QxButton-content css-m3eosc-QxButton-content"
    >
      Button
    </div>
  </button>
</DocumentFragment>
`;

exports[`Button style params loading true 1`] = `
<DocumentFragment>
  <button
    class="QxBaseButton QxBaseButton_size_medium QxBaseButton_color_brand QxBaseButton_type_contained QxBaseButton_borderRadius_medium QxBaseButton_loading QxButton QxButton_size_medium QxButton_loading css-5l479g-QxBaseButton-root-QxButton-root"
    data-testid="testStyleParams-loading-true"
    disabled=""
    type="button"
  >
    <div
      class="QxButton-content css-11blzq7-QxButton-content"
    >
      Button
    </div>
    <div
      class="QxLoader QxLoader_size_medium QxBaseButton-loader css-uhsai0-QxLoader-root-QxBaseButton-loader"
    >
      <div
        class="QxLoader-dot css-1c9k1a7-QxLoader-dot"
      />
      <div
        class="QxLoader-dot css-1c9k1a7-QxLoader-dot"
      />
      <div
        class="QxLoader-dot css-1c9k1a7-QxLoader-dot"
      />
    </div>
  </button>
</DocumentFragment>
`;

exports[`Button style params size large 1`] = `
<DocumentFragment>
  <button
    class="QxBaseButton QxBaseButton_size_large QxBaseButton_color_brand QxBaseButton_type_contained QxBaseButton_borderRadius_medium QxButton QxButton_size_large css-lnw46y-QxBaseButton-root-QxButton-root"
    data-testid="testStyleParams-size-large"
    type="button"
  >
    <div
      class="QxButton-content css-m3eosc-QxButton-content"
    >
      Button
    </div>
  </button>
</DocumentFragment>
`;

exports[`Button style params size medium 1`] = `
<DocumentFragment>
  <button
    class="QxBaseButton QxBaseButton_size_medium QxBaseButton_color_brand QxBaseButton_type_contained QxBaseButton_borderRadius_medium QxButton QxButton_size_medium css-xf29is-QxBaseButton-root-QxButton-root"
    data-testid="testStyleParams-size-medium"
    type="button"
  >
    <div
      class="QxButton-content css-m3eosc-QxButton-content"
    >
      Button
    </div>
  </button>
</DocumentFragment>
`;

exports[`Button style params size small 1`] = `
<DocumentFragment>
  <button
    class="QxBaseButton QxBaseButton_size_small QxBaseButton_color_brand QxBaseButton_type_contained QxBaseButton_borderRadius_medium QxButton QxButton_size_small css-7g7b21-QxBaseButton-root-QxButton-root"
    data-testid="testStyleParams-size-small"
    type="button"
  >
    <div
      class="QxButton-content css-59e1j2-QxButton-content"
    >
      Button
    </div>
  </button>
</DocumentFragment>
`;

exports[`Button style params size xSmall 1`] = `
<DocumentFragment>
  <button
    class="QxBaseButton QxBaseButton_size_xSmall QxBaseButton_color_brand QxBaseButton_type_contained QxBaseButton_borderRadius_medium QxButton QxButton_size_xSmall css-17m5330-QxBaseButton-root-QxButton-root"
    data-testid="testStyleParams-size-xSmall"
    type="button"
  >
    <div
      class="QxButton-content css-m3eosc-QxButton-content"
    >
      Button
    </div>
  </button>
</DocumentFragment>
`;

exports[`Button style params type contained 1`] = `
<DocumentFragment>
  <button
    class="QxBaseButton QxBaseButton_size_medium QxBaseButton_color_brand QxBaseButton_type_contained QxBaseButton_borderRadius_medium QxButton QxButton_size_medium css-xf29is-QxBaseButton-root-QxButton-root"
    data-testid="testStyleParams-type-contained"
    type="button"
  >
    <div
      class="QxButton-content css-m3eosc-QxButton-content"
    >
      Button
    </div>
  </button>
</DocumentFragment>
`;

exports[`Button style params type outlined 1`] = `
<DocumentFragment>
  <button
    class="QxBaseButton QxBaseButton_size_medium QxBaseButton_color_brand QxBaseButton_type_outlined QxBaseButton_borderRadius_medium QxButton QxButton_size_medium css-mx96a7-QxBaseButton-root-QxButton-root"
    data-testid="testStyleParams-type-outlined"
    type="button"
  >
    <div
      class="QxButton-content css-m3eosc-QxButton-content"
    >
      Button
    </div>
  </button>
</DocumentFragment>
`;

exports[`Button style params type text 1`] = `
<DocumentFragment>
  <button
    class="QxBaseButton QxBaseButton_size_medium QxBaseButton_color_brand QxBaseButton_type_text QxBaseButton_borderRadius_medium QxButton QxButton_size_medium css-5orh4g-QxBaseButton-root-QxButton-root"
    data-testid="testStyleParams-type-text"
    type="button"
  >
    <div
      class="QxButton-content css-m3eosc-QxButton-content"
    >
      Button
    </div>
  </button>
</DocumentFragment>
`;

exports[`IconButton style params borderRadius large 1`] = `
<DocumentFragment>
  <button
    class="QxBaseButton QxBaseButton_size_medium QxBaseButton_color_brand QxBaseButton_type_contained QxBaseButton_borderRadius_large QxIconButton QxIconButton_size_medium css-1p8yx8h-QxBaseButton-root-QxIconButton-root"
    data-testid="testStyleParams-borderRadius-large"
    type="button"
  >
    <div
      class="QxIconButton-icon css-5rjpdh-QxIconButton-icon"
    >
      Button
    </div>
  </button>
</DocumentFragment>
`;

exports[`IconButton style params borderRadius max 1`] = `
<DocumentFragment>
  <button
    class="QxBaseButton QxBaseButton_size_medium QxBaseButton_color_brand QxBaseButton_type_contained QxBaseButton_borderRadius_max QxIconButton QxIconButton_size_medium css-r8upnz-QxBaseButton-root-QxIconButton-root"
    data-testid="testStyleParams-borderRadius-max"
    type="button"
  >
    <div
      class="QxIconButton-icon css-5rjpdh-QxIconButton-icon"
    >
      Button
    </div>
  </button>
</DocumentFragment>
`;

exports[`IconButton style params borderRadius medium 1`] = `
<DocumentFragment>
  <button
    class="QxBaseButton QxBaseButton_size_medium QxBaseButton_color_brand QxBaseButton_type_contained QxBaseButton_borderRadius_medium QxIconButton QxIconButton_size_medium css-1d91wqo-QxBaseButton-root-QxIconButton-root"
    data-testid="testStyleParams-borderRadius-medium"
    type="button"
  >
    <div
      class="QxIconButton-icon css-5rjpdh-QxIconButton-icon"
    >
      Button
    </div>
  </button>
</DocumentFragment>
`;

exports[`IconButton style params borderRadius small 1`] = `
<DocumentFragment>
  <button
    class="QxBaseButton QxBaseButton_size_medium QxBaseButton_color_brand QxBaseButton_type_contained QxBaseButton_borderRadius_small QxIconButton QxIconButton_size_medium css-1gu38rb-QxBaseButton-root-QxIconButton-root"
    data-testid="testStyleParams-borderRadius-small"
    type="button"
  >
    <div
      class="QxIconButton-icon css-5rjpdh-QxIconButton-icon"
    >
      Button
    </div>
  </button>
</DocumentFragment>
`;

exports[`IconButton style params borderRadius xLarge 1`] = `
<DocumentFragment>
  <button
    class="QxBaseButton QxBaseButton_size_medium QxBaseButton_color_brand QxBaseButton_type_contained QxBaseButton_borderRadius_xLarge QxIconButton QxIconButton_size_medium css-nbpalj-QxBaseButton-root-QxIconButton-root"
    data-testid="testStyleParams-borderRadius-xLarge"
    type="button"
  >
    <div
      class="QxIconButton-icon css-5rjpdh-QxIconButton-icon"
    >
      Button
    </div>
  </button>
</DocumentFragment>
`;

exports[`IconButton style params borderRadius xSmall 1`] = `
<DocumentFragment>
  <button
    class="QxBaseButton QxBaseButton_size_medium QxBaseButton_color_brand QxBaseButton_type_contained QxBaseButton_borderRadius_xSmall QxIconButton QxIconButton_size_medium css-vqc25z-QxBaseButton-root-QxIconButton-root"
    data-testid="testStyleParams-borderRadius-xSmall"
    type="button"
  >
    <div
      class="QxIconButton-icon css-5rjpdh-QxIconButton-icon"
    >
      Button
    </div>
  </button>
</DocumentFragment>
`;

exports[`IconButton style params color brand 1`] = `
<DocumentFragment>
  <button
    class="QxBaseButton QxBaseButton_size_medium QxBaseButton_color_brand QxBaseButton_type_contained QxBaseButton_borderRadius_medium QxIconButton QxIconButton_size_medium css-1d91wqo-QxBaseButton-root-QxIconButton-root"
    data-testid="testStyleParams-color-brand"
    type="button"
  >
    <div
      class="QxIconButton-icon css-5rjpdh-QxIconButton-icon"
    >
      Button
    </div>
  </button>
</DocumentFragment>
`;

exports[`IconButton style params color danger 1`] = `
<DocumentFragment>
  <button
    class="QxBaseButton QxBaseButton_size_medium QxBaseButton_color_danger QxBaseButton_type_contained QxBaseButton_borderRadius_medium QxIconButton QxIconButton_size_medium css-1aordo9-QxBaseButton-root-QxIconButton-root"
    data-testid="testStyleParams-color-danger"
    type="button"
  >
    <div
      class="QxIconButton-icon css-5rjpdh-QxIconButton-icon"
    >
      Button
    </div>
  </button>
</DocumentFragment>
`;

exports[`IconButton style params color info 1`] = `
<DocumentFragment>
  <button
    class="QxBaseButton QxBaseButton_size_medium QxBaseButton_color_info QxBaseButton_type_contained QxBaseButton_borderRadius_medium QxIconButton QxIconButton_size_medium css-1w1fc8z-QxBaseButton-root-QxIconButton-root"
    data-testid="testStyleParams-color-info"
    type="button"
  >
    <div
      class="QxIconButton-icon css-5rjpdh-QxIconButton-icon"
    >
      Button
    </div>
  </button>
</DocumentFragment>
`;

exports[`IconButton style params color secondary 1`] = `
<DocumentFragment>
  <button
    class="QxBaseButton QxBaseButton_size_medium QxBaseButton_color_secondary QxBaseButton_type_contained QxBaseButton_borderRadius_medium QxIconButton QxIconButton_size_medium css-16y7zyj-QxBaseButton-root-QxIconButton-root"
    data-testid="testStyleParams-color-secondary"
    type="button"
  >
    <div
      class="QxIconButton-icon css-5rjpdh-QxIconButton-icon"
    >
      Button
    </div>
  </button>
</DocumentFragment>
`;

exports[`IconButton style params color success 1`] = `
<DocumentFragment>
  <button
    class="QxBaseButton QxBaseButton_size_medium QxBaseButton_color_success QxBaseButton_type_contained QxBaseButton_borderRadius_medium QxIconButton QxIconButton_size_medium css-we3lgt-QxBaseButton-root-QxIconButton-root"
    data-testid="testStyleParams-color-success"
    type="button"
  >
    <div
      class="QxIconButton-icon css-5rjpdh-QxIconButton-icon"
    >
      Button
    </div>
  </button>
</DocumentFragment>
`;

exports[`IconButton style params color warning 1`] = `
<DocumentFragment>
  <button
    class="QxBaseButton QxBaseButton_size_medium QxBaseButton_color_warning QxBaseButton_type_contained QxBaseButton_borderRadius_medium QxIconButton QxIconButton_size_medium css-urysa0-QxBaseButton-root-QxIconButton-root"
    data-testid="testStyleParams-color-warning"
    type="button"
  >
    <div
      class="QxIconButton-icon css-5rjpdh-QxIconButton-icon"
    >
      Button
    </div>
  </button>
</DocumentFragment>
`;

exports[`IconButton style params loading false 1`] = `
<DocumentFragment>
  <button
    class="QxBaseButton QxBaseButton_size_medium QxBaseButton_color_brand QxBaseButton_type_contained QxBaseButton_borderRadius_medium QxIconButton QxIconButton_size_medium css-1d91wqo-QxBaseButton-root-QxIconButton-root"
    data-testid="testStyleParams-loading-false"
    type="button"
  >
    <div
      class="QxIconButton-icon css-5rjpdh-QxIconButton-icon"
    >
      Button
    </div>
  </button>
</DocumentFragment>
`;

exports[`IconButton style params loading true 1`] = `
<DocumentFragment>
  <button
    class="QxBaseButton QxBaseButton_size_medium QxBaseButton_color_brand QxBaseButton_type_contained QxBaseButton_borderRadius_medium QxBaseButton_loading QxIconButton QxIconButton_size_medium QxIconButton_loading css-9awhez-QxBaseButton-root-QxIconButton-root"
    data-testid="testStyleParams-loading-true"
    disabled=""
    type="button"
  >
    <div
      class="QxIconButton-icon css-1nc6511-QxIconButton-icon"
    >
      Button
    </div>
    <div
      class="QxLoader QxLoader_size_medium QxBaseButton-loader css-uhsai0-QxLoader-root-QxBaseButton-loader"
    >
      <div
        class="QxLoader-dot css-1c9k1a7-QxLoader-dot"
      />
      <div
        class="QxLoader-dot css-1c9k1a7-QxLoader-dot"
      />
    </div>
  </button>
</DocumentFragment>
`;

exports[`IconButton style params size large 1`] = `
<DocumentFragment>
  <button
    class="QxBaseButton QxBaseButton_size_large QxBaseButton_color_brand QxBaseButton_type_contained QxBaseButton_borderRadius_medium QxIconButton QxIconButton_size_large css-1ayn32u-QxBaseButton-root-QxIconButton-root"
    data-testid="testStyleParams-size-large"
    type="button"
  >
    <div
      class="QxIconButton-icon css-5rjpdh-QxIconButton-icon"
    >
      Button
    </div>
  </button>
</DocumentFragment>
`;

exports[`IconButton style params size medium 1`] = `
<DocumentFragment>
  <button
    class="QxBaseButton QxBaseButton_size_medium QxBaseButton_color_brand QxBaseButton_type_contained QxBaseButton_borderRadius_medium QxIconButton QxIconButton_size_medium css-1d91wqo-QxBaseButton-root-QxIconButton-root"
    data-testid="testStyleParams-size-medium"
    type="button"
  >
    <div
      class="QxIconButton-icon css-5rjpdh-QxIconButton-icon"
    >
      Button
    </div>
  </button>
</DocumentFragment>
`;

exports[`IconButton style params size small 1`] = `
<DocumentFragment>
  <button
    class="QxBaseButton QxBaseButton_size_small QxBaseButton_color_brand QxBaseButton_type_contained QxBaseButton_borderRadius_medium QxIconButton QxIconButton_size_small css-v3zwcb-QxBaseButton-root-QxIconButton-root"
    data-testid="testStyleParams-size-small"
    type="button"
  >
    <div
      class="QxIconButton-icon css-5rjpdh-QxIconButton-icon"
    >
      Button
    </div>
  </button>
</DocumentFragment>
`;

exports[`IconButton style params size xSmall 1`] = `
<DocumentFragment>
  <button
    class="QxBaseButton QxBaseButton_size_xSmall QxBaseButton_color_brand QxBaseButton_type_contained QxBaseButton_borderRadius_medium QxIconButton QxIconButton_size_xSmall css-1bld30r-QxBaseButton-root-QxIconButton-root"
    data-testid="testStyleParams-size-xSmall"
    type="button"
  >
    <div
      class="QxIconButton-icon css-5rjpdh-QxIconButton-icon"
    >
      Button
    </div>
  </button>
</DocumentFragment>
`;

exports[`IconButton style params type contained 1`] = `
<DocumentFragment>
  <button
    class="QxBaseButton QxBaseButton_size_medium QxBaseButton_color_brand QxBaseButton_type_contained QxBaseButton_borderRadius_medium QxIconButton QxIconButton_size_medium css-1d91wqo-QxBaseButton-root-QxIconButton-root"
    data-testid="testStyleParams-type-contained"
    type="button"
  >
    <div
      class="QxIconButton-icon css-5rjpdh-QxIconButton-icon"
    >
      Button
    </div>
  </button>
</DocumentFragment>
`;

exports[`IconButton style params type outlined 1`] = `
<DocumentFragment>
  <button
    class="QxBaseButton QxBaseButton_size_medium QxBaseButton_color_brand QxBaseButton_type_outlined QxBaseButton_borderRadius_medium QxIconButton QxIconButton_size_medium css-1rla9lu-QxBaseButton-root-QxIconButton-root"
    data-testid="testStyleParams-type-outlined"
    type="button"
  >
    <div
      class="QxIconButton-icon css-5rjpdh-QxIconButton-icon"
    >
      Button
    </div>
  </button>
</DocumentFragment>
`;

exports[`IconButton style params type text 1`] = `
<DocumentFragment>
  <button
    class="QxBaseButton QxBaseButton_size_medium QxBaseButton_color_brand QxBaseButton_type_text QxBaseButton_borderRadius_medium QxIconButton QxIconButton_size_medium css-9scsfy-QxBaseButton-root-QxIconButton-root"
    data-testid="testStyleParams-type-text"
    type="button"
  >
    <div
      class="QxIconButton-icon css-5rjpdh-QxIconButton-icon"
    >
      Button
    </div>
  </button>
</DocumentFragment>
`;
